pennylane_qiskit/qiskit_device2.py

```diff
@@ -0,0 +1,484 @@
+
+    output_tapes = []
+    if use_sampler:
+        output_tapes.append(qml.tape.QuantumScript(tape.operations, use_sampler, shots=tape.shots))
+    if use_estimator:
+        output_tapes.append(qml.tape.QuantumScript(tape.operations, use_estimator, shots=tape.shots))
+    if other:
+        output_tapes.append(qml.tape.QuantumScript(tape.operations, other, shots=tape.shots))
+
+    return tuple(output_tapes), null_postprocessing
+
+@transform
+def validate_measurement_types(
+        tape: qml.tape.QuantumTape,
+    ) -> (Sequence[qml.tape.QuantumTape], Callable):
+
+    measurement_types = set(type(mp) for mp in tape.measurements)
+
+    if measurement_types.issubset({ExpectationMP, VarianceMP}):
+        return (tape,), null_postprocessing
+    elif measurement_types.issubset({CountsMP}):
+        return (tape,), null_postprocessing
+    else:
+        if measurement_types.intersection({CountsMP, ExpectationMP, VarianceMP}):
+            raise RuntimeError("Bad measurement combination")
+
+    return (tape,), null_postprocessing
+
+
+
+
+class QiskitDevice2(Device):
+    r"""Abstract Qiskit device for PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``).
+        backend (str | Backend): the initialized Qiskit backend
+
+    Keyword Args:
+        shots (int or None): number of circuit evaluations/random samples used
+            to estimate expectation values and variances of observables.
+        use_primitives(bool): whether or not to use Qiskit Primitives and the Qiskit Runtime Session. Defaults to False.
+    """
+    name = "Qiskit PennyLane plugin"
+    pennylane_requires = ">=0.30.0"
+    version = __version__
+    plugin_version = __version__
+    author = "Xanadu"
+
+    _operation_map = {**QISKIT_OPERATION_MAP, **QISKIT_OPERATION_INVERSES_MAP}
+
+    operations = set(_operation_map.keys())
+    observables = {
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "Identity",
+        "Hadamard",
+        "Hermitian",
+        "Projector",
+    }
+
+    short_name = "qiskit.remote2"
+
+    def __init__(self, wires, backend, shots=1024, use_primitives=False, **kwargs):
+        super().__init__(wires=wires, shots=shots)
+
+        self._backend = backend
+
+        self.runtime_service = QiskitRuntimeService(channel="ibm_quantum")
+        self._use_primitives = use_primitives
+
+        # Keep track if the user specified analytic to be True
+        if shots is None:
+            # Raise a warning if no shots were specified for a hardware device
+            warnings.warn(f"The analytic calculation of expectations, variances and probabilities "
+                          f"is only supported on statevector backends, not on the {backend}. Such "
```

Comment:
Will this work now that `backend` is not a string?

Reply:
Ah, good point, this needs to be `backend.name`